### ISSUE TITLE
Consolidate axis `bandPosition` and `tickExtra`

### ISSUE THREAD

**Issue by kanitw** [vega/vega#2131]
(Porting from https://github.com/vega/vega-lite/issues/5517)

The current design of `bandPosition` and `tickExtra` are unnecessarily complex.  
And this confused both @haldenl and I.

For context, we tried to use `bandPosition` to move band to the side and recognize that a tick is missing.  We then thought this was a bug.   I only remembered that there is `tickExtra` property an hour later.  I then thought about adding reference to `tickExtra` in the `bandPosition` docs so people will discover it. 

However, thinking more, it seems an overkill that users need to discover / learn about the `tickExtra` property because user intention would be either having the tick in the middle or on the sides (which requires one extra tick).  Requiring users to use both `bandPosition` and `tickExtra` just to do one action (having ticks on the extents of the bands) introduces a gulf of execution.  

AFAIK, the only sensible way to use these two properties is to set `bandPosition` to `1` and `tickExtra` to `true` like this:

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28}, {"a": "B", "b": 55}, {"a": "C", "b": 43},
      {"a": "D", "b": 91}, {"a": "E", "b": 81}, {"a": "F", "b": 53},
      {"a": "G", "b": 19}, {"a": "H", "b": 87}, {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal", "axis": {
      "bandPosition": 1,
      "tickExtra": true
    }},
    "y": {"field": "b", "type": "quantitative"}
  }
}
```

Setting bandPosition to `0` and `tickExtra` to `true` doesn't work as tickExtra always add a tick at the start of the axis.   

Thus, I don't see why one would ever set `bandPosition` to a value other than 1 (and 0.5, but 0.5 is already the default).  

## Potential Solutions 

I think we should simplify this by doing one of the following solutions:

a) hiding `tickExtra` and make `bandPosition` accepts either `"center"` or `"extent"` (which automatically enables "tickExtra").  We can make still  `bandPosition` take number for backward compatibility reason. 

b) We could deprecate both `tickExtra` and `bandPosition`.  We can then add a new `tickPosition` (or call it `tickBandPosition`) that takes either `"center"` or `"extent"`.
Basically, currently bandPosition only modifies `tickPosition` anyway, so it might make more sense to call it `tick(Band)Position`.

**Comment by jheer** [comment]
I have a candidate PR for this (#2135). I'm still not 100% sure if we want to commit to this route. Though it works, there are [some inelegant bits](https://github.com/vega/vega/pull/2135/files#diff-c394a8d620b04918abde197d16e55b44R54) in order to maintain backwards compatibility. In any case, I invite you to review it and share feedback!

Also, note that these updates do not address the pre-existing issue that interval-style (`"extent"`) ticks do not work well if there is any innerPadding on the band scale. Otherwise, we get ticks like this:

![image](https://user-images.githubusercontent.com/202799/68338689-544b0400-00e3-11ea-939d-cc964e76d59c.png)

To get proper ticks and also support flexible inner and outer padding, we would need to implement new scale functionality. In the image above the bars fill the bands (as defined by the band scale implementation) and the rest is empty space. We might want a different notion in which a band has a total extent including padding (for ticks) and an inner extent (for bars)? Or, we might not want to go down this road for the time being...

**Comment by jheer** [comment]
Added in v5.8.0.
